src/components/shared/SubscriberInfo/SubscriberInfo.js

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from 'react';
+import { InjectAppServices } from '../../../services/pure-di';
+import { Loading } from '../../Loading/Loading';
+import { FormattedMessage, FormattedDate } from 'react-intl';
+import { getSubscriberStatusCssClassName } from '../../../utils';
+import { StarsScore } from '../../shared/StarsScore/StarsScore';
+import SafeRedirect from '../../SafeRedirect';
+
+const SubscriberInfo = ({ email, dependencies: { dopplerApiClient } }) => {
+  const [stateSubscriber, setStateSubscriber] = useState({ loading: true });
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setStateSubscriber({ loading: true });
+      const responseSubscriber = await dopplerApiClient.getSubscriber(email);
+      if (responseSubscriber.success) {
+        const subscriber = {
+          ...responseSubscriber.value,
+          firstName: responseSubscriber.value.fields.find((x) => x.name === 'FIRSTNAME'),
+          lastName: responseSubscriber.value.fields.find((x) => x.name === 'LASTNAME'),
+        };
+
+        setStateSubscriber({
+          subscriber: subscriber,
+          loading: false,
+        });
+      }
+    };
+    fetchData();
+  }, [dopplerApiClient, email]);
+
+  if (!stateSubscriber.loading && !stateSubscriber.subscriber) {
+    return <SafeRedirect to="/Lists/MasterSubscriber/" />;
+  }
```

Comment:
Esto me parece medio raro. Se supone que `SubscriberInfo` es un componente que muestra datos de un suscriptor, teoriacamente podríamos empotrar ese componente en diferentes páginas.

Cuando el suscriptor no existe se genera esa situación: `!stateSubscriber.loading && !stateSubscriber.subscriber`, y el componente decide redirigir todo la navegación.

Si bien no es totalmente incorrecto, creo que rompe el [principio de la mínima sorpresa](https://en.wikipedia.org/wiki/Principle_of_least_astonishment). Osea, si yo meto este componente dentro de otro, no me podría imaginar que eso pueda generar una navegación.

Creo que la forma de resolver esto es mover la responsabilidad de obtener los datos fuera del componente, cualquier cosa lo hablamos.

Reply:
I agree with this. Other way could be perhaps show a message that there is no such subscriber.

Reply:
Perhaps as it said andres we need to start divinding data components from presentationals, here you can follow some examples https://krasimir.gitbooks.io/react-in-patterns/content/chapter-06/